e2e/accounts/account_node_test.go

```diff
@@ -0,0 +1,647 @@
+package accounts
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/e2e"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/node"
+	"github.com/status-im/status-go/geth/params"
+	. "github.com/status-im/status-go/testing"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	tmpDirName        = "status-tmp"
+	account1          = "test-account1.pk"
+	account2          = "test-account2.pk"
+	accountEIP55      = "test-account1-before-eip55.pk"
+	incorrectAddress  = "incorrect_address"
+	incorrectPassword = "incorrect_password"
+)
+
+func TestAccountsNodeTestSuite(t *testing.T) {
+	suite.Run(t, new(AccountsNodeTestSuite))
+}
+
+type AccountsNodeTestSuite struct {
+	e2e.NodeManagerTestSuite
+	config *params.NodeConfig
+}
+
+func (s *AccountsNodeTestSuite) SetupSuite() {
+	require := s.Require()
+
+	// setup node config
+	config, err := e2e.MakeTestNodeConfig(params.RinkebyNetworkID)
+	require.NoError(err)
+	require.NotNil(config)
+
+	// make sure that whisper is enabled
+	require.True(config.WhisperConfig.Enabled)
+
+	s.config = config
+}
+
+func (s *AccountsNodeTestSuite) SetupTest() {
+	require := s.Require()
+	s.NodeManager = node.NewNodeManager()
+	require.NotNil(s.NodeManager)
+}
+
+func (s *AccountsNodeTestSuite) BeforeTest(suiteName, testName string) {
+	require := s.Require()
+
+	// setup empty key store
+	emptyKeyStoreDir, err := ioutil.TempDir(os.TempDir(), tmpDirName)
+	require.NoError(err)
+	require.NotNil(emptyKeyStoreDir)
+
+	// update config
+	s.config.KeyStoreDir = emptyKeyStoreDir
+
+	//@NOTE(rgeraldes) account1 file will not be present in the keystore
+	// import accounts
+	require.NoError(common.ImportTestAccount(emptyKeyStoreDir, account2))
+
+	// start node
+	nodeStarted, err := s.NodeManager.StartNode(s.config)
+	require.NoError(err)
+	require.NotNil(nodeStarted)
+	<-nodeStarted
+}
+
+func (s *AccountsNodeTestSuite) AfterTest(suiteName, testName string) {
+	require := s.Require()
+
+	// stop node
+	nodeStopped, err := s.NodeManager.StopNode()
+	require.NoError(err)
+	require.NotNil(nodeStopped)
+	<-nodeStopped
+
+	require.NoError(os.RemoveAll(s.config.KeyStoreDir))
+}
+
+func (s *AccountsNodeTestSuite) TestCreateAccount() {
+	require := s.Require()
+
+	// setup account manager
+	accountManager := account.NewManager(s.NodeManager)
+	require.NotNil(accountManager)
+
+	// test cases
+	testCases := []struct {
+		context       string
```

Comment:
We usually use `name` in table-driven tests — it's essentially just a name of the case, not the "context", which is much broader term.

Reply:
@boris Is this tests yours or just legacy? If not i think it is out of scope to review them.

Reply:
@JekaMas Yep. e2e tests was written into https://github.com/status-im/status-go/pull/409. I've just moved it from geth/account to e2e/account.

@divan I think we should add it to styleguide to avoid same again.